### PR TITLE
Removed binary formatter

### LINKED ISSUE #1779: Removal of BinaryFormatter in .NET 8.0
### Description

Due to [security issues](https://aka.ms/binaryformatter), `BinaryFormatter` [will be removed in .NET 8.0](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#net-8-nov-2023) and its `Serialize` and `Deserialize` methods are currently obsolete in .NET 5.0.

There is currently a single use of `BinaryFormatter`'s `Serialize` and `Deserialize` methods:
https://github.com/fluentassertions/fluentassertions/blob/834a2db00a60e2626655a28a129c084831b85880/Src/FluentAssertions/ObjectAssertionsExtensions.cs#L142-L153

Although it's a long way off until `BinaryFormatter` is removed, I figured it'd be a good idea to bring some attention to it. I stumbled upon this while working on #1754.

### PR DESCRIPTION
**Pull request by lg2de**
This fixes #1779.

The `BinaryFormatter` is deprecated. So, it will not get part of the planned net8 target.
~~For all other targets it has been marked as obsolete, which is a breaking change.~~
After disussion in the linked issue, the support is finally removed completely.

## IMPORTANT 

* [x] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [x] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by dennisdoomen** [comment]
I thought we wanted to completely get rid of it, right @jnyrup

**Review by Corniel — APPROVED** [review]
LGTM

**Comment by lg2de** [comment]
> I thought we wanted to completely get rid of it, right @jnyrup

From [BinaryFormatter Obsoletion Strategy](https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md):
_Applications which target .NET Framework 2.x, .NET Framework 4.x, or .NET Core 3.1 are not affected by this proposal. BinaryFormatter will continue to work as expected in those applications._

So, why removing code for supported targets?

**Review by dennisdoomen — APPROVED** [review]

File: Src/FluentAssertions/ObjectAssertionsExtensions.cs [hunk 1]
```diff
@@ -12,71 +12,6 @@ namespace FluentAssertions;
...
-    /// </summary>
-    /// <param name="assertions"></param>
-    /// <param name="because">
-    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
-    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
-    /// </param>
-    /// <param name="becauseArgs">
-    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
-    /// </param>
-    public static AndConstraint<ObjectAssertions> BeBinarySerializable(this ObjectAssertions assertions, string because = "",
```
Comment (jnyrup) [inline Src/FluentAssertions/ObjectAssertionsExtensions.cs:27]: Remove example from https://github.com/fluentassertions/fluentassertions/blob/develop/docs/_pages/basicassertions.md

Comment (lg2de) [reply Src/FluentAssertions/ObjectAssertionsExtensions.cs:27]: done

**Review by jnyrup — APPROVED** [review]

**Pull request merged**